src/io/backend_cpal.rs

```diff
@@ -135,59 +244,14 @@ impl StreamConfigsBuilder {
     /// define requested hardware buffer size
     #[allow(clippy::needless_pass_by_value)]
     fn with_latency_hint(&mut self, latency_hint: AudioContextLatencyCategory) {
-        let render_quantum_size = u32::try_from(RENDER_QUANTUM_SIZE).unwrap();
-
-        // at 44100Hz sample rate (this could be even more relaxed):
-        // Interactive: 128 samples is 2,9ms
-        // Balanced:    512 samples is 11,6ms
-        // Playback:    1024 samples is 23,2ms
-        let buffer_size = match latency_hint {
-            AudioContextLatencyCategory::Interactive => render_quantum_size,
-            AudioContextLatencyCategory::Balanced => render_quantum_size * 4,
-            AudioContextLatencyCategory::Playback => render_quantum_size * 8,
-            // buffer_size is always positive and truncation is the desired behavior
-            #[allow(clippy::cast_sign_loss)]
-            #[allow(clippy::cast_possible_truncation)]
-            AudioContextLatencyCategory::Custom(latency) => {
-                if latency <= 0. {
-                    panic!(
-                        "RangeError - Invalid custom latency: {:?}, should be strictly positive",
-                        latency
-                    );
-                }
-
-                let buffer_size = latency as u32 * self.prefered.sample_rate.0;
-                buffer_size.next_power_of_two()
-            }
-        };
+        let buffer_size = super::buffer_size_for_latency_category(
+            latency_hint,
+            self.prefered.sample_rate.0 as f32,
+        ) as u32;
 
         let clamped_buffer_size: u32 = match self.supported.buffer_size() {
             SupportedBufferSize::Unknown => buffer_size,
-            // try to find the best power of two candidate in given range
```

Comment:
I would have kept this part, it seemed to me that alsa was very happy to crash when the requested `buffer_size` was not a power of two (but this probably needs more rigorous testing...)

Reply:
Interesting, but at this point, `buffer_size` is a power of two (1, 4, or 8 times 128).
The only thing now is we clamp it to `(min, max)`. It would be strange if either `min` or `max` is then not supported, no? Because that is what ALSA reports as being available.
If you could at any time have a look at the crash, that would be very beneficial!

Reply:
> Interesting, but at this point, buffer_size is a power of two (1, 4, or 8 times 128).

 Yup (or any power of two if custom latency has been given)

> The only thing now is we clamp it to (min, max). It would be strange if either min or max is then not supported, no? Because that is what ALSA reports as being available.

Yup, I agree that appears a bit weird, but if I remember well (which is maybe not the case :) the min / max values reported by alsa I have saw were quite strange

> If you could at any time have a look at the crash, that would be very beneficial!

Yup, I will try to have a closer and more precise look, I was actually hardcoding values for testing so maybe I'm wrong here